File: caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <stdlib.h>
#include <ctype.h>

int main( int argc, string argv[] )
{
    // make sure command line inputs 2 arguments only
    if ( argc != 2)
    {
       printf( "Usage: ./caesar k\n");
       return 1;
    }

    // convert string in element 2 of argv to int
    int k = atoi(argv[1]);

    // get string from user
    printf("plaintext: ");
    string text = get_string();

    // print ciphertext
    printf("ciphertext: ");

    // iterate over each character in text
    for ( int n = 0, l = strlen(text); n < l; n++ )
    {
        char oldchar = text[n];

        //encipher uppercase characters
        if (isupper(oldchar))
        {
            int oindex = oldchar - 65;
            char nindex = (oindex + k) % 26;
            char newchar = nindex + 65;
            printf("%c", newchar);
        }

        //encipher lowercase characters
        else if (islower(oldchar))
        {
            int oindex = oldchar - 97;
            char nindex = (oindex + k) % 26;
            char newchar = nindex + 97;
            printf("%c", newchar);
        }

        //print non-alphabetical characters as is
        else
        {
            printf("%c", oldchar);
        }
    }

    // final step
    printf("\n");
    return 0;
}

File: caesar.py
import sys
import cs50

# if number of args not 2, remind correct usage and stop program
if len(sys.argv) != 2:
    print("Usage: ./caesar k")
    sys.exit(1)

# convert user inputted key into int
k = int(sys.argv[1])

# get plaintext
ptext = cs50.get_string("plaintext: ")

# print 'ciphertext'
print("ciphertext: ", end = "")

# for each char in plaintext
for char in ptext:
    # if it's alphabet
    if char.isalpha() == True:
        # if it's uppercase
        if char.isupper() == True:
            # convert to int, -64 to get new index, add key and mod, then add back 64 for ascii
            print(chr((((ord(char) - 64) + k) % 26) + 64), end = "")
        # if it's lowercase
        elif char.islower() == True:
            # same procedure as uppercase but now for lower
            print(chr((((ord(char) - 96) + k) % 26) + 96), end = "")
    # if it's not alphabet, print as it is
    else:
        print(char, end = "")

# print new line
print()


